### DISCUSSION TITLE
Move autentication settings from a global scope to a per config scope

### DISCUSSION THREAD

**Discussion by NTFSvolume** [jbsparrow/CyberDropDownloader#98] [Ideas]
Basically, each config will have their own `authentication.yaml` file.

When you choose a config, you will also change the autentication file to use for the current run. 

Recently simpcity enabled scraping on the website without login in to reduce server load. However, a few threads still require you to be logged in to access them. With this change, you would be able to use a config with no login information for most cases. If you want to download from a restricted thread, just switch configs.

To remain backwards compatible, it could be implemented as an override file. Values present in the config specific `authentication.yaml` would override the global `authentication.yaml`

**Comment by jbsparrow** [comment]
I like the idea of still being able to have a global authentication file. I'll take a look at implementing this.

**Comment by jbsparrow** [comment]
Opened a PR to add this feature.
